### PR TITLE
Andrew King

### PR DESCRIPTION
**Pull request by devAndrewK**

### CODE REVIEW

**Review by philmccarthy — CHANGES_REQUESTED** [review]
Hi @devAndrewK,

There are a couple things missing that needs to be addressed before your project is complete:

## To do

- [ ] final_prep refactoring exercise (first section)
- [ ] `section2/ exercises` from https://github.com/turingschool/mod-0-project-be/tree/main/section2#if-statement-and-conditional-lessons are missing. Please confirm you went through these exercises. It's ok if you don't have the files to add.

Submit updates by EOD tomorrow (Wed 2/16).

## Feedback

Other than those 2 things, your project work looks really good. You clearly have a solid technical foundation and I'm confident you'll be ready to tackle Mod 1. I left a few comments throughout, so take a look and feel free to ask any questions in replies.

I would've liked to see you update the READMEs throughout the project, checking off items in the markdown as you went. _Not_ doing that might have contributed to you accidentally skipping a couple sections?

**Comment by devAndrewK** [comment]
As far as I could tell, I didn't see anything that could be factored further in section 1.

File: section3/exercises/hashes.rb [hunk 1]
```diff
@@ -90,12 +112,40 @@
...
 # good to have some exposure before going into Mod 1.
 
 
-# YOU DO: Create an array of at least 3 EMAIL Hashes, using the same 
+# YOU DO: Create an array of at least 3 EMAIL Hashes, using the same
 # key-value pairs you used in your email Hash above.
 # Then, print the email Array to the terminal.
+
+emails = [
+{
```
[the remark this answers is not in the document]
Comment (devAndrewK) [reply section3/exercises/hashes.rb:126]: So the main problem was the inconsistent indentation of the key-value pairs?

File: section3/exercises/hashes.rb [hunk 2]
```diff
@@ -39,16 +52,25 @@
 # Declare a variable that stores hash. Each key should be an attribute of an email and each
 # value should be some appropriate value for that key. Work to have at least 5 key-value pairs.
 
-# Write code that prints your email hash to the terminal.
+email =
+{
```
[the remark this answers is not in the document]
Comment (devAndrewK) [reply section3/exercises/hashes.rb:56]: Good to know, I know other languages do it that way, so I probably got the idea from somewhere else.

File: section3/exercises/arrays.rb [hunk 3]
```diff
@@ -17,50 +17,96 @@
...
 
 # YOU DO: Write code below that will print the number of elements in the array of
 # foods from above.
+puts foods.length
 
+puts
 
 # YOU DO: Write code below that uses a method to add "broccoli" to the foods array and
 # print the changed array to verify "broccoli" has been added
+def add_broccoli (array)
```
[the remark this answers is not in the document]
Comment (devAndrewK) [reply section3/exercises/arrays.rb:74]: I could use a gets.chomp to get input and .push that into the array, instead of the hard coded string.

**Comment by devAndrewK** [comment]
Also I honestly didn't make the connection between the README file being what is read from the browser. I thought the README was just a copy. I had tried doing the checkmarks but they weren't saving however I was doing it.

File: section3/exercises/hashes.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (philmccarthy) [reply section3/exercises/hashes.rb:126]: And the opening curly bracket should also be indented

**Comment by philmccarthy** [comment]
Hey @devAndrewK, the section about refactoring relates to making _any_ changes to section 1-3 exercises. Your commit history shows you haven't made any changes to those files since you completed them originally.

You can't think of any updates to make in all of those exercises now that you've learned more about Ruby, different methods, style conventions? It doesn't need to be much, but this is a required exercise.

**Comment by devAndrewK** [comment]
I actually did misinterpret what you meant by section 1; please let me know if my additional changes were sufficient.

**Comment by philmccarthy** [comment]
Thanks Andrew. Good to go.

Other than the sections you skipped, your project looked solid. I appreciated your reflections throughout Mod 0. And I wish you all the best moving forward! Good luck in Mod 1 🙂

**Pull request closed without merging**